lisa/features/infiniband.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+from lisa.feature import Feature
+from lisa.tools import Lspci, Dmesg
+
+FEATURE_NAME_INFINIBAND = "Infiniband"
+
+
+class Infiniband(Feature):
+    def enabled(self) -> bool:
+        return True
+
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_INFINIBAND
+
+    @classmethod
+    def can_disable(cls) -> bool:
+        return True
+
+    def is_over_sriov(self) -> bool:
+        lspci = self._node.tools[Lspci]
+        device_list = lspci.get_device_list()
+
+        return any("Virtual Function" in device.device_info for device in device_list)
+
+    # nd stands for network direct
+    # example SKU: Standard_H16mr
+    def is_over_nd(self) -> bool:
+        dmesg = self._node.tools[Dmesg]
+        return "hvnd_try_bind_nic" in dmesg.get_output()
```

Comment:
This implementation should be in Azure level, because the hvnd_try_bind_nic is a message from Microsoft drivers.

Reply:
Is this also true for is_over_sriov()?

Reply:
Good question. From the implementation of is_over_sriov, it looks like it's general logic. But if it's not, it should be moved into Azure too.